kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/async/ExecutorFactory.java

```diff
@@ -15,12 +15,14 @@
 public class ExecutorFactory {
   private static Executor executor;
 
-  public static synchronized Supplier<Executor> getExecutorSupplier(Config config) {
+  public static synchronized Supplier<Executor> getExecutorSupplier(
+      Config config, int numStreamThreads) {
```

Comment:
why to change constructor? we can directly read `num.stream.threads` from config. No?

Reply:
This config object which is passed is not necessarily guaranteed to be the whole kafka streams config. We can define it as subpath and pass only that, it depends on how consumer decides to do their config wiring.

Reply:
This is kafka streams framework and it was a cautious call that this config should be only under kafka streams config. This allows us to extend the behaviours like these in future too.

Discussed offline. We will change this.